### PR TITLE
bugbash fixes for lambda layers

### PR DESCRIPTION
**Pull request by akshbhu**
*Description of changes:*

* Added layer parameters in function update

* Added warning message to update function dependent on layer

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/amplify-category-function/src/provider-utils/awscloudformation/service-walkthroughs/lambdaLayerWalkthrough.ts [hunk 1]
```diff
@@ -112,6 +116,8 @@ export async function updateLayerWalkthrough(
     }
   }
   if (islayerVersionChanged) {
+    context.print.info('');
+    context.print.warning('Please update your Lambda function depending on this layer using - amplify function update');
```
Comment (jhockett) [inline packages/amplify-category-function/src/provider-utils/awscloudformation/service-walkthroughs/lambdaLayerWalkthrough.ts:120]: Maybe adding quotes would be better? cc @brene 
```js
context.print.warning('Please update your Lambda function depending on this layer using - "amplify function update"');
```

File: packages/amplify-category-function/src/provider-utils/awscloudformation/utils/layerParams.ts [hunk 2]
```diff
@@ -22,14 +22,14 @@ export enum Permission {
...
-  runtimes: FunctionRuntime[];
-  layerMetaData: LayerVersionMetadata;
-  getVersion: (version: number) => LayerVersionMetadata;
+  runtimes?: FunctionRuntime[];
+  layerMetaData?: LayerVersionMetadata;
+  getVersion?: (version: number) => LayerVersionMetadata;
   listVersions: () => number[];
-  getLatestVersion: () => number;
+  getLatestVersion?: () => number;
 }
```
Comment (jhockett) [inline packages/amplify-category-function/src/provider-utils/awscloudformation/utils/layerParams.ts:33]: Why are these fields optional now? (except `listVersions()`)

File: packages/amplify-category-function/src/provider-utils/awscloudformation/utils/layerParams.ts [hunk 3]
```diff
@@ -148,18 +148,16 @@ class LayerVersionState implements LayerVersionMetadata {
...
+  layerName: string,
+  isPush: boolean = false,
+): LayerMetadata => {
   const resourceDirPath = path.join(projectBackendDirPath, categoryName, layerName);
   if (!fs.existsSync(resourceDirPath)) {
     return undefined;
   }
   const parametersFilePath = path.join(resourceDirPath, layerParametersFileName);
-  const obj = context.amplify.readJsonFile(parametersFilePath).parameters;
+  const obj = JSON.parse(fs.readFileSync(parametersFilePath, 'utf8')).parameters;
```
Comment (jhockett) [inline packages/amplify-category-function/src/provider-utils/awscloudformation/utils/layerParams.ts:161]: why was this changed? `context.amplify.readJsonFile` is preferred.

File: packages/amplify-category-function/src/provider-utils/awscloudformation/utils/layerParams.ts [hunk 4]
```diff
@@ -60,7 +60,7 @@ export interface OrgsLayer {
   orgs: string[];
 }
 
-class LayerState implements LayerMetadata {
+export class LayerState implements LayerMetadata {
```
Comment (jhockett) [inline packages/amplify-category-function/src/provider-utils/awscloudformation/utils/layerParams.ts:63]: I don't see this export being used anywhere else in this PR. Was this intentional?

File: packages/amplify-category-function/src/provider-utils/awscloudformation/service-walkthroughs/lambdaLayerWalkthrough.ts (hunk 1 above)
Comment (brene) [reply packages/amplify-category-function/src/provider-utils/awscloudformation/service-walkthroughs/lambdaLayerWalkthrough.ts:120]: New Lambda layer version created. Any function that wants to use the latest layer version need to configure it by running - "amplify function update"

File: packages/amplify-category-function/src/provider-utils/awscloudformation/utils/layerParams.ts (hunk 3 above)
Comment (jhockett) [reply packages/amplify-category-function/src/provider-utils/awscloudformation/utils/layerParams.ts:161]: Whoops, that's because context is no longer passed in. Resolving.
